### PR TITLE
make obsolete runtime an error

### LINKED ISSUE #3558: Java 7 runtime no longer supported is a warning
It's time to upgrade it to an error

image

### PR DESCRIPTION
**Pull request by elharo**
fixes #3558
@briandealwis 
image

### CODE REVIEW

**Comment by briandealwis** [comment]
Failing test: [AppEngineWebXmlValidationTest](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/AppEngineWebXmlValidationTest.java#L61) creates a Java 8 project, removes the `<runtime>` element, so that it becomes Java 7, and then waits for build errors to disappear.

**Comment by elharo** [comment]
fixed

**Comment by elharo** [comment]
Problem might be ImportNativeAppEngineStandardProjectTest.

| STACK DUMP @ 2.002 min: testImportProjectFrom131_oldStyleContainerWithAppEngineApi(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)

File: plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java [hunk 1]
```diff
@@ -146,7 +146,6 @@ private static void updateOldContainers(IProject project) throws CoreException {
         CloudToolsEclipseProjectUpdater.updateProject(project, SubMonitor.convert(null));
 
     ProjectUtils.waitForProjects(project);
-    ProjectUtils.waitUntilNoBuildErrors(project);
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java]: We should move this check into `testImportAppEngineStandardJava8_from1_3_1`.  And perhaps incorporate a Java-7-error in `testImportAppEngineStandardJava7_from1_3_1` and `testImportProjectFrom131_oldStyleContainerWithAppEngineApi`?

https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/472424c9d545ce72098d29e1c2a5fff7e98264a6/plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/AppEngineWebXmlValidationTest.java#L63-L66

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java]: How necessary is this? We have separate tests that cover that logic already and presumably it slows down the test and makes it more brittle.

Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java]: Well this is an integration test…. `waitForProjects()` waits until the build errors have stabilized so it shouldn't add any more time.

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java]: OK, done.

File: plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java [hunk 2]
```diff
@@ -71,8 +71,9 @@ public void testImportAppEngineStandardJava7_from1_3_1() throws IOException, Cor
     project = SwtBotAppEngineActions.importNativeProject(bot, "AESv7", tempFolder.getRoot());
     assertTrue(project.exists());
 
-    updateOldContainers(project);
-
+    IStatus updateStatus = updateContainers(project);
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java]: So there's two situations that could occur: there is no obsolete-runtime error, or there are additional errors other than the obsolete-runtime error.

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java]: OK, added the test.

**Comment by elharo** [comment]
since the last commit we seem to have reproducible timeouts on Java 11 Eclipse 4.9 in testImportProjectFrom131_oldStyleContainerWithAppEngineApi

**Comment by elharo** [comment]
still failing and it's consistently Eclipse 4.9 on Java 11, nothing else

**Comment by elharo** [comment]
switching to Java 11 locally appears to reproduce this at the command line

**Pull request merged**